# Removing the Constant Features using the Manual Approach by `Mr. Harshit Dawar!`

* ***This approach is used to remove the constant features, they can be numerical as well as Categorical. There can be 2 ways to achieve it, one could be to encode the categorical variables, but it involves lot of processing. Therefore, the other approach is to use the number of unique values of each variable, & this is the one which is explained in this code! ***

In [1]:
# Importing the required Libraries!

import pandas as pd

In [2]:
# Loading the Data!

data = pd.read_csv("../ds/dataset_1.csv")

In [3]:
data.shape

(50000, 301)

In [4]:
data.head()

var_1  var_2  var_3  var_4  var_5  var_6  var_7  var_8  var_9  var_10  ...  \
0      0      0    0.0   0.00    0.0      0      0      0      0       0  ...   
1      0      0    0.0   3.00    0.0      0      0      0      0       0  ...   
2      0      0    0.0   5.88    0.0      0      0      0      0       0  ...   
3      0      0    0.0  14.10    0.0      0      0      0      0       0  ...   
4      0      0    0.0   5.76    0.0      0      0      0      0       0  ...   

   var_292  var_293  var_294  var_295  var_296  var_297  var_298  var_299  \
0      0.0        0        0        0        0        0        0      0.0   
1      0.0        0        0        0        0        0        0      0.0   
2      0.0        0        0        3        0        0        0      0.0   
3      0.0        0        0        0        0        0        0      0.0   
4      0.0        0        0        0        0        0        0      0.0   

      var_300  target  
0      0.0000       0  
1      0.0000       0  
2  67772.7216       0  
3      0.0000       0  
4      0.0000       0  

[5 rows x 301 columns]

In [5]:
# Changing all the variables to object type (Categorical Variables), to show that this approach
# works with categorical variables even!

data = data.astype("O")

In [6]:
data.dtypes

var_1      object
var_2      object
var_3      object
var_4      object
var_5      object
            ...  
var_297    object
var_298    object
var_299    object
var_300    object
target     object
Length: 301, dtype: object

In [9]:
# Extracting the Constant Features from the Dataset!

constantFeatures = [feature for feature in data.columns if data[feature].nunique() == 1]

In [10]:
constantFeatures

['var_23',
 'var_33',
 'var_44',
 'var_61',
 'var_80',
 'var_81',
 'var_87',
 'var_89',
 'var_92',
 'var_97',
 'var_99',
 'var_112',
 'var_113',
 'var_120',
 'var_122',
 'var_127',
 'var_135',
 'var_158',
 'var_167',
 'var_171',
 'var_178',
 'var_180',
 'var_182',
 'var_195',
 'var_196',
 'var_201',
 'var_212',
 'var_215',
 'var_225',
 'var_227',
 'var_248',
 'var_294',
 'var_297']

In [11]:
len(constantFeatures)

33

In [12]:
# Removing the Constant Features from the Dataset!

data.drop(labels = constantFeatures, axis = 1, inplace = True)

In [13]:
data.shape

(50000, 268)

***As you can see that the columns of the dataset are dropped from 301 to 268!, it is a significant reduction in the dimensionality!***

# Congratulations, you have understood another way/technique for the Feature Selection!